engine/table/src/main/java/io/deephaven/engine/table/impl/select/analyzers/SelectAndViewAnalyzer.java

```diff
@@ -532,6 +534,197 @@ void submit(
+
+            if (count == 0) {
+                // no work to do
+                completeAction.run();
+            }
+
+            final AtomicInteger nextIndex = new AtomicInteger(start);
+            final AtomicInteger remaining = new AtomicInteger(count);
+
+            final Runnable resumeAction = () -> {
+                // check for completion
+                if (remaining.decrementAndGet() == 0) {
+                    completeAction.run();
+                }
+            };
+
+            final Runnable task = () -> {
+                // this will run until all tasks have started
+                while (true) {
+                    int idx = nextIndex.getAndIncrement();
+                    if (idx < start + count) {
+                        // do the work
+                        action.run(idx, resumeAction);
+                    } else {
+                        // no more work to do
+                        return;
+                    }
+                }
+            };
+
+            // create multiple tasks but not more than one per scheduler thread
+            for (int i = 0; i < Math.min(count, threadCount()); i++) {
+                submit(executionContext,
+                        task,
+                        description,
+                        onError);
+            }
+        }
+
+        /**
+         * Provides a mechanism to iterate over a range of values serially using the {@link JobScheduler}. The advantage
+         * to using this over a simple iteration is the resumption callable on {@code action} that will trigger the next
+         * iterable. This allows the next iteration to de delayed until dependendat asynchronous serial or parallel
+         * scheduler jobs have completed.
+         *
+         * @param executionContext the execution context for this task
+         * @param description the description for
+         * @param start the integer value from which to start iterating
+         * @param count the number of times this task should be called
+         * @param action the task to perform, the current iteration index and a resume Runnable are parameters
+         * @param completeAction this will be called when all iterations are complete
+         * @param onError error handler for the scheduler to use while iterating
+         */
+        @FinalDefault
+        default void iterateSerial(ExecutionContext executionContext, LogOutputAppendable description, int start,
+                int count, IterateResumeAction action, Runnable completeAction, Consumer<Exception> onError) {
+
+            if (count == 0) {
+                // no work to do
+                completeAction.run();
+            }
+
+            final AtomicInteger nextIndex = new AtomicInteger(start);
+            final AtomicInteger remaining = new AtomicInteger(count);
+
+            // no lambda, need the `this` reference to re-execute
+            final Runnable resumeAction = new Runnable() {
```

Comment:
Otherwise, there is no way to nest parallel calls.  While we want to process the windows serially, we also want to cache the needed input sources in parallel and process bucket slices in parallel.  This implementation (similar in Serial and Parallel) allows serial continuation to be triggered by a UGP or InitThreadPool task instead of blocking and limiting the threads available for computation.